scripts/coverage_diff.py

```diff
@@ -90,15 +90,13 @@ def parse_coverage_status(coverage):
     coverage: {filename: {parent_cov: float, child_cov: float, delta_cov: float, new_file: bool}}
     output: {"exit_code":int, "message":str} 
     """
+    latest_non_terminate_status = {"exit_code":0, "message":"Coverage non decreasing.", "log_level":"notice", "terminate":False}
     for fn in coverage:
-        if coverage[fn]["new_file"] and coverage[fn]["delta_cov"] <= 20:
-            # New files need to have at least 20% coverage
-            return {"exit_code":1, "message":"New files need at least 20% coverage."} 
-        elif coverage[fn]["delta_cov"] < 0 and abs(coverage[fn]["delta_cov"]) >= min(20, coverage[fn]["parent_cov"]):
-            return {"exit_code":1, "message":"Coverage decreased severely for more than 20% or a file lost coverage."} 
-        elif coverage[fn]["delta_cov"] < 0:
-            return {"exit_code":0, "message":"Coverage decreased slightly for less than 20%"} 
-    return {"exit_code":0, "message":"Coverage non decreasing."} 
+        exit_status = file_coverage_status(coverage[fn])
+        if exit_status["terminate"]:
+            return exit_status
```

Comment:
Then you won't be able to return early, so instead combine the statuses and return at the end of the function.

Reply:
Would the following high-level structure work?
1. Convert coverage to coverage status by applying `file_coverage_status()` to all items in coverage.
2. Retrieve the combined exit status from coverage status.
3. Print the messages from coverage status.

Reply:
I have combined things together, so now it will print all error message for each file separated by new line, and exit with the highest exit code returned from `file_coverage_status()`